streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -268,9 +274,13 @@ public void onPartitionsAssigned(final Collection<TopicPartition> assignment) {
             final long start = time.milliseconds();
             try {
                 if (streamThread.setState(State.PARTITIONS_ASSIGNED) == null) {
-                    return;
-                }
-                if (streamThread.assignmentErrorCode.get() == StreamsPartitionAssignor.Error.NONE.code()) {
+                    log.error("State transition from {} to PARTITIONS_ASSIGNED failed. " +
+                                  "Will skip the task initialization", streamThread.state());
+                } else if (streamThread.assignmentErrorCode.get() != StreamsPartitionAssignor.Error.NONE.code()) {
+                    log.error("Encountered assignment error during partition assignment: {}. " +
+                                  "Will skip the task initialization", streamThread.assignmentErrorCode);
```

Comment:
There are two different error codes:
 - `VERSION_PROBING`: for this case we continue and rejoin the group; also the assignment is empty
 - `INCOMPLETE_SOURCE_TOPIC_METADATA`: for this case we shut down, and this case is already handled above.

Hence, I think we don't need to log anything for this case? (Note that we log the version probing in StreamThread later and the metadata error is logged above already)

Reply:
I guess it's still useful to have a debug log here as it helps us understand why we skip the assignment.

Reply:
Well. Its double logging... Not a big fan of that. But if you insist to keep if, also fine.

Reply:
Addressed